df_auth/drf/serializers.py

```diff
@@ -126,27 +128,70 @@ class FirstLastNameSerializerMixin(serializers.Serializer):
 
 
 class SocialTokenObtainSerializer(FirstLastNameSerializerMixin, TokenCreateSerializer):
-    access_token = serializers.CharField(write_only=True)
-    provider = serializers.ChoiceField(
-        choices=[
-            (backend.name, backend.name)
-            for backend in AUTHENTICATION_BACKENDS
-            if hasattr(backend, "name")
-        ],
+    oauth_token = serializers.CharField(write_only=True, required=False)
+    oauth_token_secret = serializers.CharField(write_only=True, required=False)
+    access_token = serializers.CharField(write_only=True, required=False)
+    oauth_type = serializers.ChoiceField(
```

Comment:
I think we can compute `oauth_type` by `provider`. So, we can remove this field.

Also, looks like we don't need all content from `contants.py` then

Reply:
That can also work. I implemented this way for flexibility, making it easy to switch between different OAuth type for same provider.